esphome/components/sgp40/sgp40.cpp

```diff
@@ -77,6 +77,26 @@ void SGP40Component::setup() {
   }
 
   this->self_test_();
+
+  /* The official spec for this sensor at https://docs.rs-online.com/1956/A700000007055193.pdf
+  indicates this sensor should be driven at 1Hz. Comments from the developers at:
+  https://github.com/Sensirion/embedded-sgp/issues/136 indicate the algorithm should be a bit
+  resilient to slight timing variations so the software timer should be accurate enough for
+  this.
+
+  This block starts sampling from the sensor at 1Hz, and is done seperately from the call
+  to the update method. This seperation is to support getting accurate measurements but
+  limit the amount of communication done over wifi for power consumption or to keep the
+  number of records reported from being overwhelming.
+
+  At configuration the component can be configured to turn off optimal sampling in which
+  case the sensor will be read out with the same cadence as the update method, leading to
+  likely inacurate measurements, but possibly accurate enough for certain use cases. 
+  */
+  if (this->optimal_sampling_) {
+    ESP_LOGD(TAG, "Using optimal sampling, setting up background sampler");
+    App.scheduler.set_interval(this, "", 1000, [this](){this->update_voc_index();});
```

Comment:
```suggestion
    this->set_interval(1000, [this](){this->update_voc_index();});
```

Component already has this method

Reply:
Oh right, sorry, I'm still trying to load this code base into my mental cache